### PR TITLE
More media types

### PR DESCRIPTION
**Pull request by jnhooper**
tl;dr;
-----
DRY code and books and tv pages are now ready to roll

nitty gritty
----------
i abstracted away some of the logic for the nextjs ssg (static site generation) stuff so that we dont have to repeat ourselves as much. This happens in `data/data.ts` i'm not too happy with that location though so maybe like api/utils or something? idk we can move it later if we hate it. essentially you pass a medium string that can either 'movies' 'tv' or 'books' based on this each function will generate a function that next.js needs to build out the static site. We do this for the [id].tsx and the index.tsx file for each of the mediums (movies, tv and books).

related work
-------------
This PR is branched off of https://github.com/mattlock12/movies/pull/3 so once that gets merged in i imagine the files changed will drop a bit. This pr is really only interested in abstracting away `getStaticProps`, `getStaticPaths`,  freeing up the index.tsx and [id].tsx  to be easier to copy and paste around. we can probably even abstract _that_ away but i am still fairly new to next and wasn't sure how to do it off the top of my head

### CODE REVIEW

**Comment by mattlock12** [comment]
@jnhooper if you feel like giving me a tour I'd be interested to know what you're doing. If not, you can just merge

File: data/data.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  }
+  return staticProps;
+}
+
+export const generateMediaStaticPaths = (mediaType: mediumDir) => {
+
+  const getStaticPaths: GetStaticPaths = async () => {
+    // const markdownDirectory = path.join(process.cwd(), 'interviews');
+    // const jsonFile = path.join(markdownDirectory, 'interviews.json');
+    // const interviewJSON = JSON.parse(fs.readFileSync(jsonFile, 'utf8'));
```
Comment (mattlock12) [inline data/data.ts:57]: ?

File: data/data.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    const mediumDirectory = path.join(process.cwd(), `data/${mediaType}`)
+    const jsonFile = await fs.readFile(path.join(mediumDirectory, 'data.json'), 'utf8')
+    const mediaJson: MediaInterface[]= JSON.parse(jsonFile)
+    const paths = mediaJson.map((media) => ({
+      params:{
+        id: `${media.id}`,
+      }
+    }))
+    // const paths = interviewJSON.map(interview => ({params: {id: interview.id}}));
```
Comment (mattlock12) [inline data/data.ts:66]: ?

**Comment by mattlock12** [comment]
I'm gonna hold off a real review til #3  is merged in

**Comment by jnhooper** [comment]
> I'm gonna hold off a real review til #3 is merged in

well i can't merge #3 so feel free
